app/lib/methods/loadMissedMessages.ts

```diff
@@ -1,47 +1,136 @@
+	}
+};
 
 const getLastUpdate = async (rid: string) => {
 	try {
 		const db = database.active;
 		const subsCollection = db.get('subscriptions');
 		const sub = await subsCollection.find(rid);
-		return sub.lastOpen?.toISOString();
+		return sub.lastOpen;
 	} catch (e) {
 		// Do nothing
 	}
 	return null;
 };
 
-async function load({ rid: roomId, lastOpen }: { rid: string; lastOpen?: Date }) {
-	let lastUpdate;
+async function load({
+	rid: roomId,
+	lastOpen,
+	updatedNext,
+	deletedNext
+}: {
+	rid: string;
+	lastOpen?: Date;
+	updatedNext?: number | null;
+	deletedNext?: number | null;
+}) {
+	const { version: serverVersion } = store.getState().server;
+	if (compareServerVersion(serverVersion, 'greaterThanOrEqualTo', '7.1.0')) {
+		let lastOpenTimestamp;
+		if (lastOpen) {
+			lastOpenTimestamp = new Date(lastOpen).getTime();
+		} else {
+			const lastUpdate = await getLastUpdate(roomId);
+			lastOpenTimestamp = lastUpdate?.getTime();
+		}
+		const result = await getSyncMessagesFromCursor(roomId, lastOpenTimestamp, updatedNext, deletedNext);
+		return result;
+	}
+
+	let lastOpenISOString;
 	if (lastOpen) {
-		lastUpdate = new Date(lastOpen).toISOString();
+		lastOpenISOString = new Date(lastOpen).toISOString();
 	} else {
-		lastUpdate = await getLastUpdate(roomId);
+		const lastUpdate = await getLastUpdate(roomId);
+		lastOpenISOString = lastUpdate?.toISOString();
 	}
 	// RC 0.60.0
 	// @ts-ignore // this method dont have type
-	const { result } = await sdk.get('chat.syncMessages', { roomId, lastUpdate });
+	const { result } = await sdk.get('chat.syncMessages', { roomId, lastUpdate: lastOpenISOString });
 	return result;
 }
 
-export function loadMissedMessages(args: { rid: string; lastOpen?: Date }): Promise<void> {
-	return new Promise(async (resolve, reject) => {
-		try {
-			const data = await load({ rid: args.rid, lastOpen: args.lastOpen });
-			if (data) {
-				const { updated, deleted }: { updated: ILastMessage[]; deleted: ILastMessage[] } = data;
-				// @ts-ignore // TODO: remove loaderItem obligatoriness
-				await updateMessages({ rid: args.rid, update: updated, remove: deleted });
+export async function loadMissedMessages(args: {
+	rid: string;
+	lastOpen?: Date;
+	updatedNext?: number | null;
+	deletedNext?: number | null;
+}): Promise<void> {
+	try {
+		const { version: serverVersion } = store.getState().server;
+
+		const data = await load({
+			rid: args.rid,
+			lastOpen: args.lastOpen,
+			updatedNext: args.updatedNext,
+			deletedNext: args.deletedNext
+		});
+		if (data) {
+			const {
+				updated,
+				updatedNext,
+				deleted,
+				deletedNext
+			}: { updated: ILastMessage[]; deleted: ILastMessage[]; updatedNext: number | null; deletedNext: number | null } = data;
+			// @ts-ignore // TODO: remove loaderItem obligatoriness
+			await updateMessages({ rid: args.rid, update: updated, remove: deleted });
+
+			if ((deletedNext || updatedNext) && compareServerVersion(serverVersion, 'greaterThanOrEqualTo', '7.1.0')) {
```

Comment:
`(deletedNext || updatedNext)` is only true when server version is >=7.1, so we can remove the version comparison.